src/crimson/os/seastore/extent_placement_manager.h

```diff
@@ -0,0 +1,193 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab expandtab
+
+#pragma once
+#include "crimson/os/seastore/cache.h"
+#include "crimson/os/seastore/cached_extent.h"
+
+namespace crimson::os::seastore {
+
+using merit_t = uint64_t;
+
+struct alloc_t {
+  paddr_t addr;
+  SegmentRef segment;
+};
+
+#define REWRITE_SEGMENT_HEADER_MAGIC "extent_placement_manager segment header v1"
+
+namespace extent_placement_manager {
+  struct segment_header_t {
```

Comment:
This is an implementation detail shared between Journal and SegmentedRewriter (which should be SegmentOOLWriter.